proc.c

```diff
@@ -628,3 +631,59 @@ procdump(void)
       cprintf("\n");
   }
 }
+
+
+int kthread_create(void (*start_func)(), void* stack){
+    //TODO
+    return 0;
+}
+
+int kthread_id(){
+    return my_thread()->tid;
+}
+
+void kthread_exit(){
+    //struct proc *curproc = myproc();
+    struct thread * curr_thread = my_thread();
+    boolean lastthread = TRUE;
+    struct proc* proc = myproc();
+    acquire(&proc->proclock);
+    struct thread * t;
+    for (t = proc->threads; t < &proc->threads[NTHREADS]; t++){
```

Comment:
maybe make it a function returning number of threads remaining, it could be useful in more cases.